slatedb/src/garbage_collector.rs

```diff
@@ -105,18 +115,10 @@ impl GarbageCollector {
     /// * `cancellation_token`: The cancellation token to use for the garbage collector.
     /// * `options`: The options for the garbage collector.
     ///
-    pub async fn start_async_task(
-        manifest_store: Arc<ManifestStore>,
-        table_store: Arc<TableStore>,
-        stats: Arc<GcStats>,
-        cancellation_token: CancellationToken,
-        options: GarbageCollectorOptions,
-        system_clock: Arc<dyn SystemClock>,
-    ) {
+    pub async fn start_async_task(&self) {
```

Comment:
I fought a ton with the lifetimes in this and `start_in_bg_thread`. I _think_ I've come to the right solution? If others know a better way,  let me know.